arch/sim/src/sim/sim_internal.h

```diff
@@ -123,6 +117,7 @@
         val[0] = flags & UINT32_MAX;                            \
         val[1] = (flags >> 32) & UINT32_MAX;                    \
                                                                 \
+        env[JB_ERRNO] = host_errno_get();                       \
```

Comment:
do you need move this change to the last patch, or merge the last one into this patch